lua/harpoon/mark.lua

```diff
@@ -103,41 +112,51 @@ M.add_file = function(file_name_or_buf_id)
 
     validate_buf_name(buf_name)
 
-    local config = harpoon.get_mark_config()
     for idx = 1, M.get_length() do
-        if config.marks[idx] == "" then
-            config.marks[idx] = buf_name
+        local filename = M.get_marked_file_name(idx)
+        if filename == "" then
+            harpoon.get_mark_config().marks[idx] = create_mark(filename)
+
             M.remove_empty_tail()
             return
         end
     end
 
-    table.insert(config.marks, buf_name)
+    table.insert(harpoon.get_mark_config().marks, create_mark(buf_name))
     M.remove_empty_tail()
 end
 
 M.remove_empty_tail = function()
     local config = harpoon.get_mark_config()
 
     for i = M.get_length(), 1, -1 do
-        if config.marks[i] ~= "" then
+        local filename = M.get_marked_file_name(i)
+        if filename ~= "" then
             return
         end
 
-        if config.marks[i] == "" then
+        if filename == "" then
             table.remove(config.marks, i)
         end
     end
 end
 
 M.store_offset = function()
-    local buf_name = get_buf_name()
-    local idx = M.get_index_of(buf_name)
-    if not M.valid_index(idx) then
-        return
-    end
+    local ok, res = pcall(function()
+        local buf_name = get_buf_name()
+        local idx = M.get_index_of(buf_name)
+        if not M.valid_index(idx) then
+            return
+        end
 
-    local line = vim.api.nvim_eval("line('.')");
+        harpoon.get_mark_config().marks[idx].row =
```

Comment:
Are `col` ever set? If it's not needed, I think it should be removed from `create_mark()`?

Reply:
I want to get col to work, I just haven't yet.  I am going to keep it in there until I figure that part out :)

Reply:
`:lua print(vim.fn.col("."))`